src/tasks/escalate.rs

```diff
@@ -16,90 +20,136 @@ pub async fn escalate_txs(app: Arc<App>) -> eyre::Result<()> {
+        for (relayer_id, txs) in txs_for_escalation {
+            futures.push(escalate_relayer_txs(&app, relayer_id, txs));
+        }
+
+        while let Some(result) = futures.next().await {
+            if let Err(err) = result {
+                tracing::error!(error = ?err, "Failed escalating txs");
+            }
         }
 
         tokio::time::sleep(app.config.service.escalation_interval).await;
     }
 }
+
+async fn escalate_relayer_txs(
+    app: &App,
+    relayer_id: String,
+    txs: Vec<TxForEscalation>,
+) -> eyre::Result<()> {
+    let relayer = app.db.get_relayer(&relayer_id).await?;
+
+    for tx in txs {
+        tracing::info!(id = tx.id, tx.escalation_count, "Escalating tx");
+
+        if !should_send_transaction(app, &relayer).await? {
+            tracing::warn!(id = tx.id, "Skipping transaction broadcast");
+
+            return Ok(());
+        }
+
+        let escalation = tx.escalation_count + 1;
+
+        let middleware = app
+            .signer_middleware(tx.chain_id, tx.key_id.clone())
+            .await?;
+
+        let fees = app
+            .db
+            .get_latest_block_fees_by_chain_id(tx.chain_id)
+            .await?
+            .context("Missing block")?;
+
+        // Min increase of 20% on the priority fee required for a replacement tx
+        let factor = U256::from(100);
+        let increased_gas_price_percentage =
+            factor + U256::from(20 * (1 + escalation));
+
+        let max_fee_per_gas_increase = tx.initial_max_fee_per_gas.0
+            * increased_gas_price_percentage
+            / factor;
+
+        let max_fee_per_gas =
+            tx.initial_max_fee_per_gas.0 + max_fee_per_gas_increase;
+
+        let max_priority_fee_per_gas =
+            max_fee_per_gas - fees.fee_estimates.base_fee_per_gas;
+
+        tracing::warn!(
+            "Initial tx fees are max = {}, priority = {}",
+            tx.initial_max_fee_per_gas.0,
+            tx.initial_max_priority_fee_per_gas.0
+        );
+        tracing::warn!("Escalating with max fee = {max_fee_per_gas} and max priority = {max_priority_fee_per_gas}");
+
+        let eip1559_tx = Eip1559TransactionRequest {
+            from: None,
+            to: Some(NameOrAddress::from(Address::from(tx.tx_to.0))),
+            gas: Some(tx.gas_limit.0),
+            value: Some(tx.value.0),
+            data: Some(tx.data.into()),
+            nonce: Some(tx.nonce.into()),
+            access_list: AccessList::default(),
+            max_priority_fee_per_gas: Some(max_priority_fee_per_gas),
+            max_fee_per_gas: Some(max_fee_per_gas),
+            chain_id: Some(tx.chain_id.into()),
+        };
+
+        let pending_tx = middleware
+            .send_transaction(TypedTransaction::Eip1559(eip1559_tx), None)
+            .await;
+
+        let pending_tx = match pending_tx {
+            Ok(pending_tx) => {
+                tracing::info!(?pending_tx, "Tx sent successfully");
```

Comment:
It would be helpful to have `tx.id` here for tracking a transaction through it's lifecycle both for investigating issues but also for measuring "time to land" onchain.